Common/src/geometry_structure.cpp

```diff
@@ -6806,6 +6782,153 @@ void CPhysicalGeometry::CompleteCommsAll(int nSends,
+  
+}
+
+void CPhysicalGeometry::SortAdjacency(CConfig *config) {
+  
+#ifdef HAVE_MPI
+#ifdef HAVE_PARMETIS
+  
+  if ((rank == MASTER_NODE) && (size > SINGLE_NODE))
+    cout << "Executing the partitioning functions." << endl;
+  
+  unsigned long iPoint, jPoint, local_size;
+  vector<unsigned long>::iterator it;
+  vector<unsigned long> adjacency_vector;
+  
+  /*--- Post process the adjacency information in order to get it into the
+   CSR format before sending the data to ParMETIS. We need to remove
+   repeats and adjust the size of the array for each local node. ---*/
+  
+  if ((rank == MASTER_NODE) && (size > SINGLE_NODE))
+    cout << "Building the graph adjacency structure." << endl;
+  
+  /*--- We can already create the array that indexes the adjacency. ---*/
+  
+  xadj_size = nPoint_Lin[rank]+1;
+  xadj      = new idx_t[xadj_size];
+  xadj[0]   = 0;
+  
+  /*--- Here, we transfer the adjacency information from a multi-dim vector
+   on a node-by-node basis into a single vector container. First, we sort
+   the entries and remove the duplicates we find for each node, then we
+   copy it into the single vect and clear memory from the multi-dim vec. ---*/
+
+  unsigned long total_adj_size = 0;
+  for (iPoint = 0; iPoint < nPoint; iPoint++) {
+    
+    /*--- For each point, sort the adjacency in ascending order
+     so that we can remove duplicates and complete the size for
+     unique set of adjacent nodes for that point. ---*/
+    
+    sort(adj_nodes[iPoint].begin(), adj_nodes[iPoint].end());
+    it = unique(adj_nodes[iPoint].begin(), adj_nodes[iPoint].end());
+    local_size = it - adj_nodes[iPoint].begin();
+    adj_nodes[iPoint].resize(local_size);
+    total_adj_size += local_size;
+  }
+  /*--- reserve allocated space to improve push_back behavior
+  in the following loop */
+  adjacency_vector.reserve(total_adj_size);
+
+  for (iPoint = 0; iPoint < nPoint; iPoint++){
+    local_size = adj_nodes[iPoint].size();
+    /*--- Move the sorted adjacency into a 1-D vector for all
+     points in order to make creating the copy for ParMETIS easier. ---*/
+    
+    for (jPoint = 0; jPoint < local_size; jPoint++) {
+      adjacency_vector.push_back(adj_nodes[iPoint][jPoint]);
+    }
+    
+    /*--- Increment the starting index for the next point (CSR). ---*/
+    
+    xadj[iPoint+1] = xadj[iPoint] + local_size;
+    
+    /*--- Free vector memory as we go. ---*/
+    
+    vector<unsigned long>().swap(adj_nodes[iPoint]);
+
+  }
+  
+  /*--- Force free the entire old multi-dim. adjacency vector. ---*/
+  
+  vector< vector<unsigned long> >().swap(adj_nodes);
+
+  /*--- Now that we know the size, create the final adjacency array. This
+   is the array that we will feed to ParMETIS for partitioning. ---*/
+  
+  adjacency_size = xadj[nPoint_Lin[rank]];
+  adjacency      = new idx_t[adjacency_size];
+
+  for (iPoint = 0; iPoint < adjacency_size; iPoint++)
+    adjacency[iPoint] = (idx_t)adjacency_vector[iPoint];
```

Comment:
In looking at the code more closely, is this an exact copy of the adjacency vector? In the case I tested `adjacency_size` was always equal to the size of the `adjacency_vector`. If this is the case maybe `adjacency` can be filled directly?

Reply:
Good question.
adjacency use ` idx_t` type while adjacency_vector use `unsigned long` .  I am not sure if filling directly will not raise issue somewhere else in the code
@economon what is your opinion on this ?

Reply:
I know it is a bit redundant, but I kept them separate bc the idx_t type size can be changed in the ParMETIS header separately from SU2 (we have actually done this recently to make it larger). It's basically for flexibility in case we want to change again.

Reply:
But since adjacency vector is a temporary what difference would it make to go directly from `adj_nodes` to `adjacency` casting to `idx_t` as you go?